commons/src/main/java/com/envyful/api/player/AbstractEnvyPlayer.java

```diff
@@ -72,28 +71,29 @@ public <A extends Attribute<B>, B> A getAttributeNow(Class<A> attributeClass) {
     }
 
     @Override
-    public void invalidateAttribute(Attribute<?> attribute) {
-        this.attributes.remove(attribute.getClass());
-    }
+    @SuppressWarnings("unchecked")
+    public <A extends Attribute<B, T>, B, C extends EnvyPlayer<T>> void setAttribute(A attribute) {
+        if (attribute == null) {
+            return;
+        }
 
-    @Override
-    public <A extends Attribute<B>, B> CompletableFuture<A> loadAttribute(
-            Class<? extends A> attributeClass, B id) {
-        AttributeInstance<A, B> instance = new AttributeInstance<>(this.saveManager.loadAttribute(attributeClass, id));
-        this.attributes.put(attributeClass, instance);
-        return instance.getAttribute();
+        if (attribute instanceof PlayerAttribute) {
+            ((PlayerAttribute<?, C, T>) attribute).setParent((C) this);
+        }
+
+        this.attributes.put(attribute.getClass(), new AttributeInstance<>(attribute));
     }
```

Comment:
I'm not sure I understand the question? I guess what you're asking is how would you load the attribute, and then set it on the player?

So the answer to that would be that you'd specifically use the AttributeTriggers rather than doing that manually. That way you don't need any additional logic surrounding loading the attribute data.